src/main/java/ch/unisg/ics/interactions/wot/td/clients/UriTemplate.java

```diff
@@ -0,0 +1,145 @@
+package ch.unisg.ics.interactions.wot.td.clients;
+
+import ch.unisg.ics.interactions.wot.td.schemas.DataSchema;
+
+import java.util.*;
+
+public final class UriTemplate {
+
+  private UriTemplate(){
+    throw new UnsupportedOperationException();
+  }
+
+  public static List<String> extract(String path){
+    List<String> extracted = new ArrayList<>();
+    int n = path.length();
+    String s = "";
+    for (int i = 0; i < n; i++){
+      if (path.charAt(i) == '{'){
+        extracted.add(s);
+        s = "{";
+      }
+      else if (path.charAt(i)=='}'){
+        s = s +"}";
+        extracted.add(s);
+        s = "";
+      }
+      else if (i==n-1){
+        s = s + path.charAt(i);
+        extracted.add(s);
+      }
+      else {
+        s = s + path.charAt(i);
+      }
+    }
+    return extracted;
+  }
+
+  public static List<String> getListVariables(String expression){
+    List<String> variables = new ArrayList<>();
+    String s = "";
+    int n = expression.length();
+    for (int i = 0; i < n;i++){
+      char c = expression.charAt(i);
+      if (c == ','){
+        variables.add(s);
+        s = "";
+      }
+      else if (i == n-1){
+        variables.add(s);
+      }
+      else if (c == '{' || c == '}' || c == '?' ){
+
+      }
+      else {
+        s = s + c;
+      }
+    }
+
+
+    return variables;
+  }
+
+  public static Set<String> getVariables(String expression){
+    return new HashSet<>(getListVariables(expression));
+  }
+
+
+
```

Comment:
These blank lines should be removed, see Google Java Style Guide for vertical spacing: https://google.github.io/styleguide/javaguide.html#s4.6-whitespace